WooCommerce/Classes/ViewRelated/Orders/Collect Payments/CollectOrderPaymentAnalytics.swift

```diff
@@ -0,0 +1,107 @@
+import Foundation
+import Yosemite
+
+final class CollectOrderPaymentAnalytics {
+
+    private let analytics: Analytics
+    private let configuration: CardPresentPaymentsConfiguration
+    private let orderDurationRecorder: OrderDurationRecorderProtocol
+    private var connectedReader: CardReader?
+    private var paymentGatewayAccount: PaymentGatewayAccount?
+
+    var connectedReaderModel: String {
```

Comment:
Does it make sense if this property is optional? That way if there is no reader connected we get a `nil` value.

Reply:
So... I defaulted it to the empty string because that's what we already do in the analytics. There are a few (receipts) which we remove the `reader_model` property if there's no reader connected, and we could change it to do that every time. I don't _think_ that would have a significant impact on anything.

Reply:
Updated in 832ca44 to track `none_connected`. It's a little tricky to test, but it can be done with the cancellation event if you turn off the reader (during connection) and then cancel on the subsequent error message.

<img width="743" alt="Tracking log" src="https://user-images.githubusercontent.com/2472348/219673925-16b27355-5ac4-4abd-ab05-7bbdffcaff54.png">